[Merged by Bors] - restructured usage guide & new landing page (#344)
stackabletech/superset-operator: docs/modules/superset/pages/usage-guide/security.adoc

= Security

== [[authentication]]Authentication

> Every user has to authenticate themselves before using Superset.

Reviewer: ```suggestion
Every user has to be authenticated before using Superset: there are several ways in which this can be set up.
```

> There are multiple options to set up the authentication of users.

Reviewer: ```suggestion
```

=== Webinterface
The default setting is to manually set up users via the Webinterface.

=== LDAP

Superset supports xref:nightly@home:concepts:authentication.adoc[authentication] of users against an LDAP server. This requires setting up an xref:nightly@home:concepts:authentication.adoc#authenticationclass[AuthenticationClass] for the LDAP server.
The AuthenticationClass is then referenced in the SupersetCluster resource as follows:

[source,yaml]
----
apiVersion: superset.stackable.tech/v1alpha1
kind: SupersetCluster
metadata:
  name: superset-with-ldap-server
spec:
  image:
    productVersion: 1.5.1
    stackableVersion: 23.4.0-rc2
  [...]
  authenticationConfig:
    authenticationClass: ldap    # <1>
    userRegistrationRole: Admin  # <2>
----

<1> The reference to an AuthenticationClass called `ldap`

> <2> The default role that all users are assigned to

Reviewer: ```suggestion
<2> The default role to which all users are assigned
```

Users that log in with LDAP are assigned to a default https://superset.apache.org/docs/security/#roles[Role] which is specified with the `userRegistrationRole` property.

You can follow the xref:nightly@home:tutorials:authentication_with_openldap.adoc[] tutorial to learn how to set up an AuthenticationClass for an LDAP server, as well as consulting the xref:nightly@home:reference:authenticationclass.adoc[] reference.

== [[authorization]]Authorization
Superset has a concept called `Roles` which allows you to grant user permissions based on roles.
Have a look at the https://superset.apache.org/docs/security[Superset documentation on Security].

=== Webinterface

> You can see all the available roles in the Webinterface of Superset.

Reviewer: I think the first line is a duplicate.
```suggestion
```
